### PR TITLE
Ahodges/SD_config_mobile

### PR DESCRIPTION
**Pull request by lamagnifica**
I definitely recommend looking at this in split view...
@nimisha, I updated the configuration instructions to match the relevant parts of the https://openedx.atlassian.net/wiki/pages/viewpage.action?spaceKey=MA&title=Settings+for+Mobile+API+Sandbox page. I'm not sure if I've characterized the differences between configuring the json vs yml files correctly. Any help appreciated. I added push notifications to the end of the mobile.rst file -- I'd like to work on providing more detail, but only if/when you or another engineer has the time to spend, or if we need more for our June delivery.
@mhoeber , I did some editing for style guide issues, and made a suggestion about the way the contents display at the top of the mobile.rst file. Let me know if this is the right direction.
DOC-1921, spec here: https://docs.google.com/document/d/1TK98LD04jOImemlrWVKkL6AMufWzPDnlGVlWA7Rob6Y/edit#heading=h.98h83c2jnbei

### CODE REVIEW

**Comment by lamagnifica** [comment]
@aleffert, please look over the info I added for the app-side configuration per your docs.

**Comment by aleffert** [comment]
@lamagnifica looks good :+1:

File: en_us/install_operations/source/mobile.rst [hunk 1]
```diff
@@ -6,54 +6,201 @@
...
 applications and customizing their Open edX installation to support their use.
 
-.. contents:: Chapter Contents:
+.. contents:: Section Contents 
+   :local:
 
-****
-Code
-****
+.. Mark, this option ^ prevents the section name from appearing as a link. What do you think about standardizing on this format at the beginning of each rst?
```
Comment (mhoeber) [inline en_us/install_operations/source/mobile.rst:15]: not sure I understand, show me tomorrow

File: en_us/install_operations/source/mobile.rst [hunk 2]
```diff
@@ -6,54 +6,201 @@
...
+=======================================
+Enable Features in ``server-vars.yml``
+=======================================
+
+You can enable many of the mobile application features in the 
+``server-vars.yml`` file. Configuration settings added to this file 
+are reapplied automatically when you update ``edx-platform``.
+   
+.. note:: Currently, you must manually add 
+          ``"OAUTH_ENFORCE_SECURE": " ",`` to the
```
Comment (mhoeber) [inline en_us/install_operations/source/mobile.rst]: ? should this value be blank?

**Comment by mhoeber** [comment]
couple of questions, looks good. :+1:  

can you add change log entry too

File: en_us/install_operations/source/mobile.rst (hunk 2 above)
Comment (lamagnifica) [inline en_us/install_operations/source/mobile.rst]: @nasthagiri or @jibsheet or @aleffert is this setting accurate?

Comment (jibsheet) [inline en_us/install_operations/source/mobile.rst]: This is likely a workaround for old ansible bugs.
I'm in the middle of turning it to false for all sandboxes, after which I think this line becomes irrelevant.
I'm running a quick test of setting it to false (Rather than "") on a newer sandbox right now.

Comment (jibsheet) [inline en_us/install_operations/source/mobile.rst]: Just tested, adding EDXAPP_OAUTH_ENFORCE_SECURE: False to server vars and it percolates down to lms.env.json properly.
This is probably still necessary on mobile3 because it has an outdated configuration, but will not be needed on newer mobile setups. @lamagnifica

**Comment by lamagnifica** [comment]
@lou-wang please give your thoughts.

File: en_us/install_operations/source/mobile.rst [hunk 3]
```diff
@@ -6,54 +6,201 @@
...
+following keys.
+
+::
+
+    PARSE:
+      NOTIFICATIONS_ENABLED: true
+      APPLICATION_ID: {your application id}
+      CLIENT_KEY: {your client key}
+
+
```
Comment (lamagnifica) [inline en_us/install_operations/source/mobile.rst:185]: per @lou-wang , no need to include the info in the remainder of this file

**Comment by lamagnifica** [comment]
@nasthagiri completely rewritten after our meeting this afternoon. Please let me know if this is accurate, complete, and using terminology/phrasing that open edX developers are likely to understand. Thanks!

**Comment by lamagnifica** [comment]
@jibsheet

File: en_us/install_operations/source/mobile.rst [hunk 4]
```diff
@@ -6,54 +6,212 @@
...
+the custom configuration file of each mobile application.
 
-In particular, you need to add the following under the features section:
+====================================
+Enable Mobile Application Features
+====================================
 
-.. code-block:: json
+.. note:: Configuration settings added to the ``lms.env.json`` file are reset 
+ to their default values each time you update edx-platform.
```
Comment (nasthagiri) [inline en_us/install_operations/source/mobile.rst]: In case this needs to be more clear:
They are reset only each time you update a sandbox or a devstack vagrant image.
The `lms.env.json` file is not in the source code, so it doesn't get updated when the edx-platform source is updated.  For example, after a `git pull`.

File: en_us/install_operations/source/mobile.rst [hunk 5]
```diff
@@ -6,54 +6,212 @@
...
 
-To configure a video module with a low resolution encoding for mobile, enter the URL to the mobile-targeted video as the first URL in the "Video File URLs" list in the video module's Advanced Editor in edX Studio.
+You create an OAuth client ID and secret that are specific to your
+installation for use by the mobile applications. The procedure that follows
+assumes that you create a different client for each of the edX mobile
+applications.
+
+To create your OAuth clients, follow these steps.
+
+#. Log in to the Django administration console for your base URL. For example,
```
Comment (nasthagiri) [inline en_us/install_operations/source/mobile.rst:80]: If they haven't already, they should create a super admin user on their server instance.
Readers can see `https://github.com/edx/configuration/wiki/edX-Managing-the-Full-Stack` for more information on managing an edX instance.

File: en_us/install_operations/source/mobile.rst [hunk 6]
```diff
@@ -6,54 +6,212 @@
...
+   ``http://{your_URL}/admin``.
+
+#. In the **Oauth2** section, select **Clients**. 
+
+#. Select **Add client**. A dialog box opens with the **Client id** and
+   **Client secret** populated for the new client.
+
+#. Enter the **Url** and the **Redirect Url** for the first application. For example,
```
Comment (nasthagiri) [inline en_us/install_operations/source/mobile.rst]: You may want to make it clear that the `Url` and `Redirect Url` settings are not used by our code.  They can enter whatever value they want just for their own accounting purposes.

File: en_us/install_operations/source/mobile.rst [hunk 7]
```diff
@@ -6,54 +6,212 @@
...
+
+   PARSE_KEYS = {

+     "APPLICATION_ID": "{app_id}",

+     "REST_API_KEY": "{API_key}"

+   }
+
+#. Save the ``edx/app/edxapp/cms.auth.json`` file.
```
Comment (nasthagiri) [inline en_us/install_operations/source/mobile.rst]: And restart the server after saving the updated file.

File: en_us/install_operations/source/mobile.rst [hunk 8]
```diff
@@ -6,54 +6,212 @@
...
+
+#. In the **Contentstore** section, select **Push notification configs**. 
+
+#. Select **Add push notification config**. 
+
+#. Verify that **Enabled** is selected, and then select **Save**.
+
+#. Restart the server.
+
+.. is a restart needed?^^
```
Comment (nasthagiri) [inline en_us/install_operations/source/mobile.rst]: Restart is needed after saving the json file, but not needed after updating the Django admin settings.  This is why devOps prefers moving most of our configuration settings to Django admin so they don't have to restart/redeploy production servers.

**Comment by nasthagiri** [comment]
:+1: lgtm overall, pending suggestions/comments above.

File: en_us/install_operations/source/mobile.rst (hunk 4 above)
Comment (jibsheet) [inline en_us/install_operations/source/mobile.rst]: Possibly something like "are reset to their default values if you use ansible to update edx-platform" - this covers sandboxes, devstack and folks who run our plays (vs using git pull)

**Comment by lamagnifica** [comment]
@jibsheet , @lou-wang

File: en_us/install_operations/source/mobile.rst [hunk 9]
```diff
@@ -6,54 +6,213 @@
...
+           ...
+           "ENABLE_MOBILE_REST_API": true,
+           "ENABLE_OAUTH2_PROVIDER": true,
+           "ENABLE_COMBINED_LOGIN_REGISTRATION": true
+       }
 
+#. Save the ``edx/app/edxapp/lms.env.json`` file.
 
-Additionally, you need to create an OAuth key and secret specific to your installation for use by the apps. You can do this by logging into the Django administration console at ``<YOUR_EDX_INSTALLATION>/admin``. From there, choose **Clients** under the **Oauth2** section. If you don't already have any clients set up, you will need to add one. Choose **Add client** and create a client. The client id and secret that you see will need to be added to your mobile app configuration. For more information about how to do that, see the documentation for each app linked above.
+#. Restart the server.
```
Comment (jibsheet) [inline en_us/install_operations/source/mobile.rst:60]: If we always say "restart the server" and folks know we mean to restart a service, I'm fine with this, but I realize some people may reboot their devstack or instance which isn't needed.  They do need to restart the service (sorry I missed this on previous edits).  On a sandbox this would be something like

/edx/bin/supervisorctl restart edxapp:\* edxapp_worker:*

If we say "restart the server" elsewhere in the docs, then I'm fine with it.

File: en_us/install_operations/source/mobile.rst [hunk 10]
```diff
@@ -6,54 +6,213 @@
...

+     "APPLICATION_ID": "{app_id}",

+     "REST_API_KEY": "{API_key}"

+   }
+
+2. Save the ``edx/app/edxapp/cms.auth.json`` file.
+
+#. Restart the server.
```
Comment (jibsheet) [inline en_us/install_operations/source/mobile.rst:181]: Same "restart the server vs service note" from above

**Comment by jibsheet** [comment]
Overall looks good, but see my note about potentially confusing restart note.

File: en_us/install_operations/source/mobile.rst (hunk 9 above)
Comment (lamagnifica) [inline en_us/install_operations/source/mobile.rst:60]: @jibsheet we say "restart the server" throughout the Installing & Configuring guide. See http://edx.readthedocs.org/projects/edx-installing-configuring-and-running/en/latest/.
That said, if they need to restart a service here instead of the server, I can easily rephrase this as "Restart the {name of service} service."

Comment (jibsheet) [inline en_us/install_operations/source/mobile.rst:60]: If "restart the server" is our convention, then ship it.  As a sysadmin I'm always sensitive to what requires an instance restart vs a service restart, but this isn't the place to go round and round about that.

**Comment by jibsheet** [comment]
:+1:
